include/nuttx/mm/map.h

```diff
@@ -193,6 +193,42 @@ void vm_release_region(FAR struct mm_map_s *mm, FAR void *vaddr,
 
 #endif
 
+#ifdef CONFIG_MM_UMAP
+
+/****************************************************************************
+ * Name: vm_map_region
+ *
+ * Description:
+ *   Allocate virtual memory and maps physical memory range into user space
+ *   of the current process.
+ *
+ * Input Parameters:
+ *   paddr - Starting physical address
+ *   size  - Size of the address range
+ *
+ * Returned Value:
+ *   Virtual address of the range, or NULL if error
+ *
+ ****************************************************************************/
+
+uintptr_t vm_map_region(uintptr_t paddr, size_t size);
```

Comment:
where is this function implemented?

Reply:
This patch is for interfaces definitions only as we are highly to change based on review discussions. The implementations will be based on `vm_alloc_region()` and `up_shmat()`. This is simply a wrapper function for device drivers to use, I am unsure if there is better place in the source tree to put them.

Reply:
but, the implementation doesn't provide in this patch? I would suggest organizing your change that one patch fix one problem completely.